pkg/trie/triedb/node.go

```diff
@@ -26,11 +30,51 @@ type (
 	}
 
 	newValueRef struct {
-		hash *common.Hash
+		hash common.Hash
 		Data []byte
 	}
 )
 
+func NewEncodedValue(value nodeValue, partial []byte, childF childFunc) (codec.EncodedValue, error) {
+	var newHashedValueHash common.Hash
+	switch v := value.(type) {
+	case newValueRef:
```

Comment:
this function has 2 switches, but I think you can merge both.

Reply:
Done! I simplified that method